### DISCUSSION TITLE
Introducing the League of Legends Desktop Application Built with Uno Platform

### DISCUSSION THREAD

**Discussion by jamesnet214** [unoplatform/uno#18751] [Show and tell]
This project was born out of a passion for exploring new possibilities in cross-platform development and sharing my experiences with the developer community.

![image](https://github.com/user-attachments/assets/c12ecb3f-44d9-420d-b366-f48f034ed50b)

![image](https://github.com/user-attachments/assets/87ec91f4-fa38-443c-9616-58c152fdcc42)

![image](https://github.com/user-attachments/assets/4d068e82-b6bc-4428-a9ad-67aa2c373e3c)

![image](https://github.com/user-attachments/assets/e400442c-b3fa-4add-8d89-14c5eb925731)

![image](https://github.com/user-attachments/assets/ca6c39b0-4c9d-4a8c-b73a-a9f722704385)

![image](https://github.com/user-attachments/assets/8ed17ae8-aed2-44dd-aded-46922d972927)

The primary goal of this project is to assist desktop developers familiar with WPF technology in beginning their journey into cross-platform development using the Uno Platform. By demonstrating how to extend the skills and experience accumulated in WPF to the Uno Platform, I aim to support developers in making a smooth transition from desktop applications to cross-platform applications.

Since 2008, I have been researching and gaining experience with WPF technology. Over the years, various cross-platform technologies such as Xamarin, MAUI, Uno Platform, AvaloniaUI, and OpenSilver have evolved, enabling us to extend the skills acquired in WPF to the Uno Platform.

This **League of Legends** Uno Platform version is built upon the rich user experience of XAML-based design from WPF, object-oriented design principles of C#, and a systematic project structure. It is a new challenge following my previous **League of Legends** WPF version.

This project demonstrates how to implement large-scale projects on the Uno Platform. By implementing complex controls as **CustomControls**, I aim to provide developers with valuable learning resources. Additionally, it includes various technical implementations using Uno, showcasing how to utilize the powerful features of the Uno Platform in real-world projects.

In particular, this project presents a comprehensive approach to distributed design for large-scale applications. It demonstrates how to structure and manage complex applications through modularized structures, efficient state management, and scalable architecture.

The **Jamesnet.Core** framework library is designed based on .NET Standard 2.0 to function identically in both WPF and Uno. This library is provided as **Jamesnet.Window** for WPF and **Jamesnet.Uno** for the Uno Platform.

In this project, I directly reference the actual source code of **Jamesnet.Core** and **Jamesnet.Uno**, allowing you to learn about the design methodologies of XAML-based frameworks.

Additionally, this repository has been introduced on the .NET Foundation. Please refer to the [.NET Foundation's LinkedIn post](https://www.linkedin.com/posts/dotnetfoundation_we-are-thrilled-to-spotlight-jamesnet214-activity-7242528138071531520-S64n?utm_source=share&utm_medium=member_desktop) for more details.

[![ReadMe Card](https://github-readme-stats.vercel.app/api/pin/?username=jamesnetgroup&repo=leagueoflegends-uno)](https://github.com/jamesnetgroup/leagueoflegends-uno)

[![ReadMe Card](https://github-readme-stats.vercel.app/api/pin/?username=jamesnetgroup&repo=leagueoflegends-wpf)](https://github.com/jamesnetgroup/leagueoflegends-wpf)

**Comment by DevTKSS** [comment]
Hello @jamesnet214,
could you maybe tell more about the settings like which template would be used for this, which markup and maybe also extensions? Which parts of the applications would you tell do profit most from the things uno provides except of just "run on web" which is as far as I know in wpf only with one of the view Controls possible, but very limited. You did mention the use of .Net core instead of the actual lowest .Net 6 in templates which made me wonder about the setup you might have used.
Would be very interested in more informations since I might have to use .net framework compartible things for one of my future applications so there would be the only choose between winforms (please don't ;) ) and the here mentioned wpf. Maybe u used the UnoIslands, maybe not...?
If that might be the wrong place to speak about, feel free to sent me a DM here on Github :)

Thank you for your great example you did show above! Will have a look into it in the next time/days for learning purposes :) You did a great job an that. I did see a LOL repo example on some Youtube Videos mentioned but did not have the time to have a look until now. Really cool if such thing would/is possible doing with uno!
Kind regards
